ctapipe/tools/tests/test_merge.py

```diff
@@ -122,3 +124,24 @@ def test_allowed_tels(tmp_path, dl1_file, dl1_proton_file):
 
     s = SubarrayDescription.from_hdf(output)
     assert s.tel.keys() == {1, 2}
+
+
+def test_split_datasets_by(tmp_path, dl1_file, dl1_proton_file):
```

Comment:
You don't actually validate anything here, you just check that the command runs without problem.

Please add checks that validate that the output file is indeed how it should be. E.g. by opening it with the DL1 event source and iterating over the first couple events.

Reply:
Please see above, I suggested checking the total length of the tables. However, the DL1 files are too small to run proper checks as pointed out by @HealthyPear. How should we proceed?

Reply:
Can we use other files? But then it's a general issue which we can solve after this PR.

We should have files that contain at least some triggered events.